packages/flutter/test/material/icon_button_test.dart

```diff
@@ -2770,6 +2771,51 @@ void main() {
 
       expect(tester.takeException(), isNull);
     });
+
+    testWidgetsWithLeakTracking('IconButton should dispose its internal MaterialStatesController', (WidgetTester tester) async {
+      Widget pageBuilder(BuildContext context) {
+        return Scaffold(
+          appBar: AppBar(title: const Text('AppBar')),
+          body: Builder(builder: (BuildContext context) {
+            return Center(
+              child: ElevatedButton(
```

Comment:
Seems we still didn't use `IconButton` here. I think the example I provided above might be enough to show there's no leaking.